### DISCUSSION TITLE
does It ngx.var works in phases of kong plugin?

### DISCUSSION THREAD

**Discussion by zffocussss** [Kong/kong#9433] [Help] [answered]
PDK does not support it?
as I want to log some additonal variables which should be generated by setting ngx.var on the access phase of kong plugins.

**Comment by narate** [comment, marked as answer]
I test with `pre-function: config.log` = `"local kong=kong; local ngx=ngx; return kong.log(\"host: \", ngx.var.host)"`  it's working fine. Custom plugin should work

Ref:
## ngx.var.VARIABLE

syntax: ngx.var.VAR_NAME

context: set_by_lua*, rewrite_by_lua*, access_by_lua*, content_by_lua*, header_filter_by_lua*, body_filter_by_lua*, log_by_lua*, balancer_by_lua*
https://github.com/openresty/lua-nginx-module#ngxvarvariable

**Reply by StarlightIbuki** [reply]
In addition to that, you can use `kong.nginx.get_ctx` and `kong.nginx.set_ctx` to access Nginx variables in external plugins.
